test/unit-test.el

```diff
@@ -229,3 +230,78 @@ then run BODY."
    "def bar /* foo */"
    (search-forward "foo")
    (should (not (memq 'font-lock-string-face (faces-at-point))))))
+
+(ert-deftest groovy-highlight-variable-assignment ()
```

Comment:
Emacs is all about accommodating different people's preferences :)

A defcustom sounds like a good idea. Something like `groovy-highlight-assignments`?